app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -242,19 +247,106 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 	 * not authorized then the the [QueueMetadata] is created from the app remote API.
 	 */
 	fun createLikedSongsQueueMetadata() {
+		if (createQueueMetadataJob?.isActive == true) {
+			createQueueMetadataJob?.cancel()
+		}
+
 		createQueueMetadataJob = GlobalScope.launch(defaultDispatcher) {
 			queueItems = webApi.getLikedSongs(this@SpotifyAppController) ?: emptyList()
+
 			if (queueItems.isNotEmpty()) {
-				queueMetadata = QueueMetadata("Liked Songs", null, queueItems)
-				loadQueueCoverart()
+				queueMetadata = QueueMetadata(queueTitle, null, queueItems)
+				val likedSongsTemporaryPlaylistStateKey = AppSettings.KEYS.SPOTIFY_LIKED_SONGS_PLAYLIST_STATE
+				val likedSongsStateJson = appSettings[likedSongsTemporaryPlaylistStateKey]
+				val temporaryPlaylistState = if (likedSongsStateJson.isBlank()) {
+					Log.d(TAG, "No previous liked songs state found.")
+					createTemporaryPlaylistState(SpotifyWebApi.LIKED_SONGS_PLAYLIST_NAME, likedSongsTemporaryPlaylistStateKey)
+				} else {
+					Log.d(TAG, "Found previous liked songs state.")
+					loadTemporaryPlaylistState(likedSongsStateJson, likedSongsTemporaryPlaylistStateKey)
+				}
+				if (temporaryPlaylistState == null) {
+					Log.e(TAG, "Error creating liked songs playlist, falling back to app remote API")
+					queueItems = emptyList()
+					createQueueMetadata(PlayerContext(queueUri, queueTitle, null, null))
+					return@launch
+				}
+
+				saveTemporaryPlaylistState(temporaryPlaylistState, likedSongsTemporaryPlaylistStateKey)
 
 				callback?.invoke(this@SpotifyAppController)
 			} else {
-				createQueueMetadata(PlayerContext(queueUri, "Liked Songs", null, null))
+				createQueueMetadata(PlayerContext(queueUri, queueTitle, null, null))
 			}
 		}
 	}
 
+	/**
+	 * Creates a [TemporaryPlaylistState] for the supplied playlist name, creating the playlist if it
+	 * is not present and adding the queueItems in context to it. If the playlist creation fails
+	 * then null is returned.
+	 */
+	private suspend fun createTemporaryPlaylistState(playlistName: String, appSettingsTemporaryPlaylistStateKey: AppSettings.KEYS): TemporaryPlaylistState? {
+		val queueItemsHashCode = queueItems.hashCode().toString()
+		val existingPlaylistUri = webApi.getPlaylistUri(playlistName)
+		val playlistUri: String
+		val playlistId: String
+		if (existingPlaylistUri == null) {
+			Log.d(TAG, "No user playlist for $playlistName found, creating a new one.")
+			val uri = webApi.createPlaylist(playlistName, L.MUSIC_TEMPORARY_PLAYLIST_DESCRIPTION) ?: return null
+			webApi.addSongsToPlaylist(uri.id, queueItems)
+			playlistUri = uri.uri
+			playlistId = uri.id
+		} else {
+			Log.d(TAG, "User playlist for $playlistName found, using existing playlist.")
+			playlistUri = existingPlaylistUri.uri
+			playlistId = existingPlaylistUri.id
+		}
+
+		val temporaryPlaylistState = TemporaryPlaylistState(queueItemsHashCode, playlistUri, playlistId, null)
+
+		queueUri = temporaryPlaylistState.playlistUri
```

Comment:
I believe this is unused

Reply:
Please see this comment: https://github.com/hufman/AndroidAutoIdrive/pull/263#discussion_r690780026